hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml

```diff
@@ -1637,6 +1637,23 @@
   </description>
 </property>
 
+<property>
+  <name>dfs.client.failover.cache-active.enabled</name>
+  <value>true</value>
+  <description>
+    Whether caching active namenode index is enabled.
+  </description>
+</property>
+
+<property>
+  <name>dfs.client.failover.cache-active.dir</name>
+  <value>${java.io.tmpdir}</value>
```

Comment:
hadoop.tmp.dir may be a good alternative.

Reply:
Thanks for reviewing, hadoop.tmp.dir was the default directory considering at the beginning, but later we found there may be a lot of cache files (seventy or eighty in our production environment), which will eventually cause hadoop.tmp.dir to look messy. Hadoop.tmp.dir has a neat directory structure originally, and I think we shoud keep it simple and tidy. How do you think?

Reply:
In our environment, a HDFS client can dynamically load the core-site.xml and hdfs-site.xml 
which corresponding to the target hdfs uri (this is an enhancement we made ourselves). In this case, there will be a lot of  cache files ( As there are many target HDFS clusters). 

This kind of dynamic loading mechanism is still relatively common in a large production environment.  So I think it's better to put these cache files in a less noticeable directory.

Reply:
i'm ok. thanks for the details.